File: lib/news.js
import NewsAPI from 'newsapi'

export default class News {
    constructor(key) {
        if(!key) {
            throw new Error('News API Key Missing')
        }
        
        this.newsAPI = new NewsAPI(key).v2

    }

    async headlines() {
        if(!this.newsAPI) {
            throw new Error('news API cannot be defined')
        }

        let response = await this.newsAPI.topHeadlines({
            sources: 'fox-news, cnn'
        })
    
        return response
    }

    async headlinesBySource(source) {
        if(!this.newsAPI) {
            throw new Error('news API cannot be defined')
        }

        let response = await this.newsAPI.topHeadlines({
            sources: source
        })
    
        return response
    }

}

File: fairnews/news.py
# Pull news from sources

# Setup News API
from newsapi import NewsApiClient

class News:
    def __init__(self, api_key):
        self.newsapi = NewsApiClient(api_key)
    
    def headlines(self):
        headlines = self.newsapi.get_top_headlines(sources='cnn, fox-news')
        return(headlines)